buildsystem/DetectProjectVersion.cmake

```diff
@@ -1,21 +1,25 @@
-# Copyright 2015-2016 the openage authors. See copying.md for legal info.
+# Copyright 2015-2019 the openage authors. See copying.md for legal info.
 
 # Determines the version of the project and stores it in the variable PROJECT_VERSION.
 
 # If the project version was already defined or previously retrieved, don't bother.
 if(PROJECT_VERSION)
+	message(STATUS "Project version already set")
 	return()
 endif()
 
 if(IS_DIRECTORY "${CMAKE_SOURCE_DIR}/.git")
 	# if .git exists, try running git describe
 	execute_process(
-		COMMAND git describe
+		COMMAND git describe --tag --abbrev=0
```

Comment:
This will just give `v0.4.0` without the commit and all this for the project command of cmake. That is beautiful, because it sets all the version commands (patch, minor, etc.) from numbers separated by dots. Is there any problem with not having the commit id?

Reply:
Ok, so as we make use right now of the more sophisticated project-Command for version setting, we could use another variable where we save the commit-id and count for compiling it in. The project-version-command `takes a <version> argument composed of **non-negative integer components**, i.e. <major>[.<minor>[.<patch>[.<tweak>]]], and sets the variables ...` if you propose a variable name I could change the behaviour that we make use of the project command in the right way **AND** can hardcore our commit id und commit count into the game.